### PR TITLE
add DarkLightToggle

### PR DESCRIPTION
**Pull request by vibros68**
This pull request add DarkLightToggle, that will replace the old simple
toggle as the disccussion from [politeiagui issue #2497](https://github.com/decred/politeiagui/issues/2497)

Screenshot from 2021-09-16 23-32-27

This pr upgrade `react-spring` to the newer version. So please upgrade
your dependency before testing

### CODE REVIEW

File: src/theme/lightTheme.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 const lightTheme = {
   "color-primary": "#2970ff",
-  "color-primary-dark": "#091440",
+  "color-primary-dark": "var(--color-blue-darkest)",
```
Comment (amass01) [inline src/theme/lightTheme.js:3]: Hrmm don't change theme variables values this affects so many other places also in `pigui`.

I guess you want to create new theme variables which are specific to your components needs
then use these in css styling to apply custom colors.

**Review by amass01 — CHANGES_REQUESTED** [review]
Needs some love

File: src/theme/darkTheme.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ const darkTheme = {
   "comment-censored-background": "var(--dimmed-card-background)",
   "comment-new-background": "var(--color-primary-darker)",
 
-  "text-color": "#99C1E3",
+  "text-color": "var(--color-blue-darker)",
```
Comment (amass01) [inline src/theme/darkTheme.js:41]: Same here

File: src/components/Toggle/DarkLightToggle.jsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useSpring, animated } from "react-spring";
+import { useTheme, getThemeProperty } from "../../theme";
+
```
Comment (amass01) [inline src/components/Toggle/DarkLightToggle.jsx:5]: extra empty line

**Review by amass01 — CHANGES_REQUESTED** [review]
As mentioned on matrix: 
```
just noticed `src/components/Toggle/DarkLightToggle.jsx`  is wrong - you are using the 
native toggle styling file, your component is a new component which needs a separate 
new dir.
also, please add the new component to the docz and some basic tests like all other components.
```

**Review by amass01 — CHANGES_REQUESTED** [review]
Hitting the following when visiting docz DarkLightToggle page:

image

**Review by amass01 — CHANGES_REQUESTED** [review]
Wrap the JS code inside the `.mdx` file with the following:

image

**Review by amass01 — APPROVED** [review]
LGTM

**Review by tiagoalvesdulce — APPROVED** [review]
ACK

**Pull request merged**